extensions/2.0/Khronos/KHR_texture_astc/README.md

```diff
@@ -0,0 +1,261 @@
+    },
+    "extensionsUsed": [
+        "KHR_texture_astc"
+    ],
+    "textures": [
+        {
+            "source": 0,
+            "extensions": {
+                "KHR_texture_astc": {
+                    "source": 1
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "mimeType": "image/png",
+            "bufferView": 1
+        },
+        {
+            "mimeType": "image/ktx2",
+            "bufferView": 2
+        }
+    ]
+}
+```
+
+### Using Without a Fallback
+
+To use KTX v2 image with ASTC compression without a fallback, define `KHR_texture_astc` in both `extensionsUsed` and `extensionsRequired`. The `texture` object will then have its `source` property omitted as shown below.
+
+```json
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "extensionsUsed": [
+        "KHR_texture_astc"
+    ],
+    "extensionsRequired": [
+        "KHR_texture_astc"
+    ],
+    "textures": [
+        {
+            "extensions": {
+                "KHR_texture_astc": {
+                    "source": 0
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "uri": "astc_image.ktx2"
+        }
+    ]
+}
+```
+
+### JSON Schema
+
+[texture.KHR_texture_astc.schema.json](schema/texture.KHR_texture_astc.schema.json)
+
+## KTX v2 Images with Astc compression 
+
+To cover a broad range of use cases, this extension allows different ASTC block sizes as well as LDR, HDR and 3D images. These can be determined from the vkFormat of the KTX image.
+
+For the purposes of this extension, the following texture types are defined:
+
+- **RGB:** A texture that uses red, green, and blue channels. Alpha channel is either assumed to be always equal to `1.0` (`255`) or unused and not sampled at runtime. Such textures MAY be encoded with the sRGB transfer function.
+  > **Note:** RGB textures from the core glTF 2.0 specification include:
+  > - `baseColorTexture` (opaque case)
+  > - `normalTexture`
+  > - `emissiveTexture`
+  > - `metallicRoughnessTexture` (when combined with `occlusionTexture`, otherwise red channel is ignored)
+
+- **RGBA:** A texture that uses all four channels. Such textures MAY be encoded with the sRGB transfer function (not applied to Alpha).
+  > **Note:** RGBA textures from the core glTF 2.0 specification include:
+  > - `baseColorTexture` (non-opaque case)
+
+- **Red:** A texture that uses only red channel. All other channels are unused and their values are not sampled at runtime. Such textures MUST NOT be encoded with the sRGB transfer function.
+  > **Note:** Red textures from the core glTF 2.0 specification include:
+  > - `occlusionTexture` (standalone)
+
+- **Luminance-Alpha:** A texture that uses only Luminance and Alpha channels. To sample these textures use `.ga` swizzle in your shaders. Such textures MAY be encoded with sRGB transfer function.
+  > **Note:** The core glTF 2.0 specification has no examples of luminance-alpha textures.
```

Comment:
@lexaknyazev Thanks for your contribution.

Can I just confirm you mean to remove the swizzle bits? or the whole section? I can remove the `.ga` swizzle requirements since it can be any of `.ra`, `.ga` or `.ba`. What I would like is to have some way to represent an `RG` texture type which is usually a Luminance-Alpha texture.

There is already precendence for this in the KHR_texture_basisu and I am tempted to just put that back in.

```
- **Red-Green:** A texture that uses only red and green channels. Blue and Alpha channels are unused and their values are not sampled at runtime. Such textures MUST NOT be encoded with sRGB transfer function.
  > **Note:** The core glTF 2.0 specification has no examples of red-green textures.
```

Reply:
The Basis extension explicitly mentions Red and Red-Green textures because Basis ETC1S codec needs special handling during transcoding of such textures.

Since ASTC decoder always provides defined values for all four channels regardless of the compressed data, all of these subsections (RGB, RGBA, ...) are not needed for ASTC.

Reply:
Fixed in https://github.com/KhronosGroup/glTF/pull/2369/commits/8781d2296877799387f293c4ebc3da6bd0b9a254